prow/pod-utils/clone/clone.go

```diff
@@ -128,8 +128,9 @@ func (g *gitCtx) commandsForBaseRef(refs prowapi.Refs, gitUserName, gitUserEmail
 	if cookiePath != "" {
 		commands = append(commands, g.gitCommand("config", "http.cookiefile", cookiePath))
 	}
-	commands = append(commands, g.gitCommand("fetch", g.repositoryURI, "--tags", "--prune"))
-	commands = append(commands, g.gitCommand("fetch", g.repositoryURI, refs.BaseRef))
+
+	commands = append(commands, g.gitCommand("fetch", g.repositoryURI, "--tags", "--prune", "--depth", strconv.Itoa(refs.CloneDepth)))
```

Comment:
We need to pass `--depth` only when it's non-zero:

```
$ git clone git@github.com:kubernetes/test-infra.git --depth 0
fatal: depth 0 is not a positive number
```

Reply:
great catch. will take care of it.